packages/typescript-client/src/client.ts

```diff
@@ -207,53 +212,60 @@ export class ShapeStream {
       }
 
       try {
-        await this.fetchClient(url.toString(), { signal })
-          .then(async (response) => {
-            if (!response.ok) {
-              throw await FetchError.fromResponse(response, url.toString())
-            }
+        const response = await this.fetchClient(url.toString(), { signal })
```

Comment:
I love the `await` refactoring here! One improvement that would be nice to get in to make iteration over this easier maybe would be to make error handling a bit more careful? I mean to only wrap the `fetch` call in a try-catch (and maybe inside a separate function?) and to only handle fetch errors there, so that we're not blindly retrying on failure to parse a type or smth like that